src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -420,4 +420,14 @@ public function testVeryLongQuotedStrings()
 
         $this->assertEquals($longStringWithQuotes, $arrayFromYaml['longStringWithQuotes']);
     }
+
+    public function testBooleanMappingKeysAreConvertedToStrings()
+    {
+        $this->assertSame(array('false' => 'foo'), Inline::parse('{false: foo}'));
```

Comment:
> I missed that part. In this case, I would merge these tests into the 2.7 branch and deprecate the behaviour in master.

:+1:

> Changing this to throw exceptions for any non integer or string keys is a big BC break, and I'm not sure this one would actually benefit users that much.

It is more for interop with other languages, in symfony `false: ` will give a string but in other languages, it could be a real bool, especially when using the yaml encoder of the Serializer component as it can be a file you know nothing about.